src/compute.ts

```diff
@@ -1,5 +1,33 @@
 import { Game } from "./types";
 
 export function compute(game: Game): number {
-  throw new Error("Not yet implemented"); // TODO
+    // Variable declarations
+    let score: number = 0;
+
+    // Loop over all of the frames and calculate bonusScores
+    for (let i = 0; i < game.length; i++) {
+        const frame = game[i];
+        const frameScore = game[i].reduce((a, b) => a + b, 0);
+
+        // Strike: score 10 points + the number of pins you knock down for the entire next frame.
+        if (frameScore === 10) {
+            // Get the next frame and the next next frame
+            // || [] is a fallback in case the next frame is undefined
+            const nextFrame = game[i + 1] || [];
```

Comment:
Nitpick prefer `??` over `||`. `??` checks whether it is undefined or null while `||` checks if it is falsy, this often gives bugs when working with strings (`""`) or numbers (`0`), though here it will work because it are objects but it's a good habbit.